### PR TITLE
feat: add evt listeners for SetAlwaysOnTop

### PR DESCRIPTION
**Pull request by codebytere**
##### Description of Change

Resolves https://github.com/electron/electron/issues/12466.

Adds new event listener for `SetAlwaysOnTop`: **"always-on-top-change"**

/cc @ckerr @MarshallOfSound 

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: Add new event listener for window `SetAlwaysOnTop`

### CODE REVIEW

File: atom/browser/api/atom_api_top_level_window.cc [hunk 1]
```diff
@@ -262,6 +262,14 @@ void TopLevelWindow::OnWindowLeaveHtmlFullScreen() {
   Emit("leave-html-full-screen");
 }
 
+void TopLevelWindow::OnWindowEnterAlwaysOnTop() {
+  Emit("enter-always-on-top");
```
[the remark this answers is not in the document]
Comment (codebytere) [reply atom/browser/api/atom_api_top_level_window.cc]: hmm...i see what you mean 🤔 

is there an alt nomenclature you'd prefer to see?

[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply atom/browser/api/atom_api_top_level_window.cc]: * `always-on-top-changed`
* `is-always-on-top` / `not-always-on-top` 🤔 

I prefer the first one

[the remark this answers is not in the document]
Comment (ckerr) [reply atom/browser/api/atom_api_top_level_window.cc]: :+1: for `always-on-top-changed` with an arg, similar to what we do for `accessibility-support-changed`

File: docs/api/browser-window.md [hunk 2]
```diff
@@ -542,6 +542,10 @@ Emitted when the window enters a full-screen state triggered by HTML API.
 
 Emitted when the window leaves a full-screen state triggered by HTML API.
 
+#### Event: 'always-on-top-change' _macOS_
+
+Emitted when the window is set or unset to show always on top of other windows.
```
Comment (MarshallOfSound) [inline docs/api/browser-window.md]: This needs a block like below

```markdown
Returns:

* `event` Event
* `isAlwaysOnTop` Boolean

Emitted when...
```

File: atom/browser/api/atom_api_top_level_window.cc [hunk 3]
```diff
@@ -262,6 +262,10 @@ void TopLevelWindow::OnWindowLeaveHtmlFullScreen() {
   Emit("leave-html-full-screen");
 }
 
+void TopLevelWindow::OnWindowAlwaysOnTopChange() {
+  Emit("always-on-top-change", IsAlwaysOnTop());
```
Comment (ckerr) [inline atom/browser/api/atom_api_top_level_window.cc]: s/change/changed for consistency with `accessibility-support-changed`, `accent-color-changed`, `cursor-changed`, etc.

**Pull request merged**